source-mysql/discovery.go

```diff
@@ -16,19 +16,28 @@ import (
 )
 
 func (db *mysqlDatabase) DiscoverTables(ctx context.Context) (map[string]*sqlcapture.DiscoveryInfo, error) {
+	var tableMap = make(map[string]*sqlcapture.DiscoveryInfo)
```

Comment:
you intend to eventually get rid of this separate query over tables?

instead of introducing a separate query for tables, could you instead update the columns query to return `table_type` (as you have already done below) and then condition on the column's table type?

Reply:
I started that way initially, but the column-discovery helper function currently returns a list of _column_ information objects and adding an "is this a column of a base table" flag to that felt worse. And since we're doing filtering on both the type _and_ the names of tables now, it felt cleaner to just divide up the responsibilities correctly -- now the column-discovery logic just lists every column of every table, and the table-discovery logic decides what tables we're interested in.

Reply:
To be clear, I think that it makes sense to keep table discovery and column listing distinct inside the database-specific discovery code here going forward, the bits I hope to revert are the existence of the "BaseTable" flag in the resulting discovery info and the conditional logic around auto-ignoring views bound to existing captures.